### PR TITLE
Move tabIcon handling to iconHandler and rename setIcon to setInterceptorIcon

### PR DESCRIPTION
**Pull request by KillariDev**

### CODE REVIEW

**Review by MicahZoltu — APPROVED** [review]
My comment isn't a blocker because the existing code has the same bug, but the bug I described is a critical bug that needs to be fixed.

File: extension/app/ts/background/iconHandler.ts [hunk 1]
```diff
@@ -24,26 +25,61 @@ export function updateExtensionIcon(port: browser.runtime.Port) {
...
+			}
+		}
+		browser.tabs.onUpdated.addListener(listener)
+		if( (await browser.tabs.get(tabId)).status === 'complete') {
+			browser.tabs.onUpdated.removeListener(listener)
+			resolve(undefined)
+		}
+	})
+
+	await waitForLoaded
```
[the remark this answers is not in the document]
Comment (KillariDev) [reply extension/app/ts/background/iconHandler.ts]: I modified it to use Future instead

File: extension/app/ts/background/iconHandler.ts [hunk 2]
```diff
@@ -24,26 +26,68 @@ export function updateExtensionIcon(port: browser.runtime.Port) {
...
+			}
+			return waitForLoadedFuture.resolve()
+		} catch(error) {
+			if (error instanceof Error) return waitForLoadedFuture.reject(error)
+			return waitForLoadedFuture.reject(new Error('Unknown error'))
+		}
+	}
+
+	waitForLoaded()
+	await waitForLoadedFuture
```
Comment (MicahZoltu) [inline extension/app/ts/background/iconHandler.ts:79]:
```suggestion
	await waitForLoaded()
```

File: extension/app/ts/background/iconHandler.ts [hunk 3]
```diff
@@ -24,26 +26,68 @@ export function updateExtensionIcon(port: browser.runtime.Port) {
...
+				return waitForLoadedFuture.resolve()
+			}
+			return waitForLoadedFuture.resolve()
+		} catch(error) {
+			if (error instanceof Error) return waitForLoadedFuture.reject(error)
+			return waitForLoadedFuture.reject(new Error('Unknown error'))
+		}
+	}
+
+	waitForLoaded()
```
Comment (MicahZoltu) [inline extension/app/ts/background/iconHandler.ts:78]: No need for the intermediate function here.  
```suggestion
	try {
		const listener = function listener(tabIdUpdated: number, info: browser.tabs._OnUpdatedChangeInfo) {
			if (info.status === 'complete' && tabId === tabIdUpdated) {
				browser.tabs.onUpdated.removeListener(listener)
				return waitForLoadedFuture.resolve()
			}
		}
		browser.tabs.onUpdated.addListener(listener)
		if( (await browser.tabs.get(tabId)).status === 'complete') {
			browser.tabs.onUpdated.removeListener(listener)
			return waitForLoadedFuture.resolve()
		}
		return waitForLoadedFuture.resolve()
	} catch(error) {
		if (error instanceof Error) return waitForLoadedFuture.reject(error)
		return waitForLoadedFuture.reject(new Error('Unknown error'))
	}
```

**Pull request merged**